update setFlags - wrap every call in runOnUiThread

stas3112rus opened the pull request:
In current verion libary we updated state application step by step. (turnScreenOn, showWhenLocked, keepScreenOn)
In this case we have a problem, when android app was killed. App can have unlock and unshown screen or shown and lock screen in the same time. 

For this problem I found solution, that we need to update state our app only with one action https://stackoverflow.com/questions/53191903/how-to-show-the-incoming-call-screen-when-the-screen-is-locked/53192049#53192049

Commit:
update setFlags params - use only one parameter for toggling state

Livan-pro commented:
Hello! I appreciate your contribution. However, there are a few issues:

1. Changing params to a single boolean flag might work for you, but it reduces flexibility. What if someone needs to set only one parameter? Instead, we can do something like this:
```typescript
export async function setFlags(
  params:
    | {
        turnScreenOn?: boolean;
        showWhenLocked?: boolean;
        keepScreenOn?: boolean;
      }
    | boolean
): Promise<void> {
  if (!isAndroid) throw new Error('Implemented only on Android!');
  await ActivityUtils.setFlags(
    typeof params === 'boolean'
      ? { turnScreenOn: params, showWhenLocked: params, keepScreenOn: params }
      : params
  );
}
```

2. I'm not sure how changing `setTurnScreenOn` and `setShowWhenLocked` to window flags fixes something. These two flags are deprecated since API 27 (Android 8.1): [FLAG_TURN_SCREEN_ON](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_TURN_SCREEN_ON), [FLAG_SHOW_WHEN_LOCKED](https://developer.android.com/reference/android/view/WindowManager.LayoutParams#FLAG_SHOW_WHEN_LOCKED). But if it really fixes something for you, let's add them as boolean flags to params map. For example, `windowTurnScreenOn` and `windowShowWhenLocked`.

stas3112rus commented:
I agree with you about the first point. 
About second, It's sad, that these settings are deprecated. They really helped us. 
Maybe we need to switch order for setTurnScreenOn, setShowWhenLocked and keepScreenOn 
I will check it and return to you with an answer.

Commit:
fix: updated ActivityUtilsModule

updated ActivityUtilsModule with not deprecated props
revert not useful changes

stas3112rus commented:
Sorry for long wait. 
I found, that problem, that `setTurnScreenOn` and `setShowWhenLocked` were used out `UiThreadUtil.runOnUiThread` 
After I put them inside, everything works well. 
I think, All worked in my previous version because `window.addFlags` and `window.clearFlags` were inside this property despite the fact flags `FLAG_TURN_SCREEN_ON` and ` FLAG_SHOW_WHEN_LOCKED` are deprecated. 

Also, I returned all changes except `ActivityUtilsModule.java` file

Livan-pro reviewed (approved):
LGTM.

Livan-pro commented:
Released as v0.1.1
